soc: arm: stm32g0: add STM32G070 SoC series

ehay64 opened the pull request:
This commit adds support for the STM32G070xx SoCs
by STMicroelectronics.

Signed-off-by: Eric Hay <EHay@sierrawireless.com>

Commit:
soc: arm: stm32g0: add STM32G070 SoC series

This commit adds support for the STM32G070xx SoCs
by STMicroelectronics.

Signed-off-by: Eric Hay <EHay@sierrawireless.com>

ehay64 commented:
Hi @galak and @erwango, do you mind having a look at this? Sorry I guess I can't add reviewers yet.

erwango commented on soc/arm/st_stm32/stm32g0/Kconfig.defconfig.stm32g071xx:
```diff
@@ -12,4 +12,12 @@ config SOC
 config NUM_IRQS
 	default 32
 
+config CLOCK_STM32_PLL_Q_DIVISOR
+	int "PLL Q Divisor"
+	depends on CLOCK_STM32_SYSCLK_SRC_PLL
+	default 2
+	range 2 8
+	help
+	  PLL Q VCO divisor, allowed values: 2-8
+
```
If some modification has to be done on clock_control driver, please split in a different commit.
Then, please keep Kconfig symbols in their own driver when doable.
In this case, this can be done as follows in `drivers/clock_control/Kconfig.stm32g0`:
```suggestion
config CLOCK_STM32_PLL_Q_DIVISOR
	int "PLL Q Divisor"
	depends on CLOCK_STM32_SYSCLK_SRC_PLL && (SOC_STM32G031XX || SOC_STM32G071XX)
	default 2
	range 2 8
	help
	  PLL Q VCO divisor, allowed values: 2-8
          Limited to STM32G0X1 variants
```

erwango reviewed (changes requested):
Thanks for this addition.
A couple of comments otherwise looks good.

Besides, g070 is likely a subset of g071.
There are some other subtleties (shared irqs fir instance), but with no impact on device tree description.
So, it would be nice to move things around at this level and get g071 including the new g070as follows:
```suggestion
-#include <st/g0/stm32g0.dtsi>
+#include <st/g0/stm32g070.dtsi>
```

Last (but no request to change), the pinmux addition will soon be deprecated due to recent introduction of dts based pinctrl configuration.

ehay64 commented:
Hi @erwango, thanks for the pointers, I'll make these changes.

Commit:
dts: arm: stm32g0: Have the stm32g071 inherit from stm32g070

The g071 is a superset of the g070. This change reflects
that relationship.

Signed-off-by: Eric Hay <EHay@sierrawireless.com>

Commit:
soc: arm: stm32g0: Set IRQs to 30 for stm32g070 series

The number of IRQs for the g070 is 30, not 32.

Signed-off-by: Eric Hay <EHay@sierrawireless.com>

Commit:
drivers: clock_control: stm32g0: Enable the Q divisor on g0X1 variants

The PLL Q divisor does not exist on stm32g0X0 variants. It should only
be configured for g0X1 variants.

Signed-off-by: Eric Hay <EHay@sierrawireless.com>

ABOSTM reviewed (approved):
LGTM
